File: tests/statistical_tests/spot_checks/fishers-exact.test.R
#!/usr/bin/env Rscript

cat("\n📊 Fisher's Exact Test\n")
cat(paste(rep("-", 80), collapse = ""), "\n")

contingency_table <- matrix(c(8, 2, 1, 5), nrow = 2, byrow = TRUE)

# Fisher's exact test
result <- fisher.test(contingency_table, alternative = "two.sided", conf.level = 0.95)

cat("Contingency Table: [[8, 2], [1, 5]]\n")
cat("Odds ratio (MLE):", result$estimate, "\n")
cat("p-value:", result$p.value, "\n")
cat("CI lower:", result$conf.int[1], "\n")
cat("CI upper:", result$conf.int[2], "\n")

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("FISHER'S EXACT TEST SPOT CHECK\n")
cat(paste(rep("=", 80), collapse = ""), "\n")

File: tests/statistical_tests/spot_checks/fishers-exact.test.ts
import { stats } from "../../../src/dataframe/mod.ts";

Deno.test("Fisher's Exact Test Spot Check", () => {
  console.log("\n📊 Fisher's Exact Test");
  console.log("-".repeat(80));

  const contingencyTable = [
    [8, 2],
    [1, 5],
  ];

  const result = stats.test.categorical.fishersExact({
    contingencyTable,
  });

  console.log("Contingency Table:", contingencyTable);
  console.log("Odds ratio (MLE):", result.effect_size.value);
  console.log("p-value:", result.p_value);
  console.log("CI:", [
    result.confidence_interval.lower,
    result.confidence_interval.upper,
  ]);
});

console.log("\n" + "=".repeat(80));
console.log("FISHER'S EXACT TEST SPOT CHECK");
console.log("=".repeat(80));
